main/src/ca/uwaterloo/flix/Main.scala

```diff
@@ -90,13 +90,19 @@ object Main {
       json = cmdOpts.json,
       output = cmdOpts.output.map(s => Paths.get(s)),
       progress = true,
-      threads = cmdOpts.threads.getOrElse(Runtime.getRuntime.availableProcessors()),
+      threads = cmdOpts.threads.getOrElse(Options.Default.threads),
       xnobooltable = cmdOpts.xnobooltable,
       xstatistics = cmdOpts.xstatistics,
       xstrictmono = cmdOpts.xstrictmono,
       xnoseteffects = cmdOpts.xnoseteffects,
       xnobooleffects = cmdOpts.xnobooleffects,
-      xvirtualthreads = cmdOpts.xvirtualthreads
+      xvirtualthreads = cmdOpts.xvirtualthreads,
+      xnooptimizer = cmdOpts.xnooptimizer,
+      incremental = Options.Default.incremental,
+      target = Options.Default.target,
+      test = Options.Default.test,
+      loadClassFiles = Options.Default.loadClassFiles,
+      xallowredundancies = Options.Default.xallowredundancies
```

Comment:
What's the point of having default if we are going to do this?

I think either:

- We remove default (forcing you always to choose)
- Or we keep the current system where we start with Options.Default.copy

Reply:
`Default` organizes the defaults well and is used as the basis for `DefaultForTesting` etc.

Reply:
Yes, but still, whats the point of not using it here?

Reply:
So that we get a compilation error if we add an option and forget to update it.

More generally, I think there are two cases we are concerned about:
1. We are specifying almost all of the options (as in this case)
2. We want the defaults except a particular option (as in the `DefaultTest` case)

For 1. it's not appropriate to use `Default` due to lack of compilation protections, but for 2. it's more appropriate since it's less likely that a change to the API would necessitate a change in `DefaultTest`.

Reply:
I am not entirely convinced, but I will take it.

Reply:
Can I then be pedantic and ask you to keep them in the same order as in options? :D